src/modals/MappingTest/components/TestMappingMappingSelect.vue

```diff
@@ -0,0 +1,238 @@
+					:clearable="false"
+					:loading="mappingsLoading || mappingTest.loading"
+					required
+					@input="emitMappingSelected">
+					<!-- eslint-disable-next-line vue/no-unused-vars vue/no-template-shadow  -->
+					<template #no-options="{ search, searching, loading }">
+						<p v-if="loading">
+							Loading...
+						</p>
+						<p v-if="!loading && !mappings.options?.length">
+							Er zijn geen mappings beschikbaar
+						</p>
+					</template>
+					<!-- eslint-disable-next-line vue/no-unused-vars  -->
+					<template #option="{ id, label, summary, removeStyle }">
+						<div :class="removeStyle !== true && 'mapping-option'">
+							<!-- custom style is enabled -->
+							<SitemapOutline v-if="!removeStyle" :size="25" />
+							<span v-if="!removeStyle">
+								<h6 style="margin: 0">
+									{{ label }}
+								</h6>
+								{{ summary }}
+							</span>
+							<!-- custom style is disabled -->
+							<p v-if="removeStyle">
+								{{ label }}
+							</p>
+						</div>
+					</template>
+				</NcSelect>
+
+				<NcButton :disabled="mappingTest.loading || !mappings.value || !inputObject.isValid"
+					type="success"
+					@click="testMapping()">
+					<template #icon>
+						<NcLoadingIcon v-if="mappingTest.loading" :size="20" />
+						<ContentSaveOutline v-if="!mappingTest.loading" :size="20" />
+					</template>
+					Test
+				</NcButton>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script>
+import {
+	NcSelect,
+	NcButton,
+	NcLoadingIcon,
+} from '@nextcloud/vue'
+
+import SitemapOutline from 'vue-material-design-icons/SitemapOutline.vue'
+import ContentSaveOutline from 'vue-material-design-icons/ContentSaveOutline.vue'
+
+export default {
+	name: 'TestMappingMappingSelect',
+	components: {
+		NcSelect,
+		NcButton,
+		NcLoadingIcon,
+	},
+	props: {
+		mappingItem: {
+			type: Object,
+			required: true,
+		},
+		inputObject: {
+			type: Object,
+			required: true,
+		},
+	},
+	data() {
+		return {
+			mappings: [],
+			mappingsLoading: false,
+			// mapping test
+			mappingTest: {
+				result: {}, // result from the testMapping function
+				success: null,
+				loading: false,
+				error: false,
+			},
+		}
+	},
+	watch: {
+		// activates when prop is received
+		mappingItem(newVal) {
+			if (!newVal?.id) return
+			this.mappings.value = this.mappings.options.find((option) => option.id === (newVal?.id || Symbol('mapping item id not found'))) || null
+		},
+		mappingTest: {
+			handler(newVal) {
+				this.$emit('mapping-test', {
+					...newVal,
+				})
+			},
+			deep: true,
+		},
+		mappingsLoading(newVal) {
+			this.$emit('mapping-selected', {
+				loading: newVal,
+			})
+		},
+	},
+	mounted() {
+		this.fetchMappings()
+	},
+	methods: {
+		emitMappingSelected(event) {
+			this.$emit('mapping-selected', {
+				selected: event,
+			})
+		},
+		fetchMappings(currentMappingItem = null) {
+			this.mappingsLoading = true
+
+			mappingStore.refreshMappingList()
+				.then(() => {
+					if (currentMappingItem) {
```

Comment:
This if statement will never fire because you set the `currentMappingItem` to `null`.
I would change it to check on `mappingStore.mappingItem`